packaging/suse/0001-patch-in-version-numbers-for-images.patch

```diff
@@ -0,0 +1,179 @@
+From: Florian Bergmann <fbergmann@suse.de>
+Date: Fri, 12 Oct 2018 17:27:28 +0200
+Subject: [PATCH] patch in version numbers for images
+
+---
+ manifests/haproxy.yaml |  2 +-
+ manifests/private.yaml |  4 ++--
+ manifests/public.yaml  | 28 ++++++++++++++--------------
+ 3 files changed, 17 insertions(+), 17 deletions(-)
+
+diff --git a/manifests/haproxy.yaml b/manifests/haproxy.yaml
+index ada52aa..24f4144 100644
+--- a/manifests/haproxy.yaml
++++ b/manifests/haproxy.yaml
+@@ -20,7 +20,7 @@ spec:
+       operator: "Exists"
+   containers:
+     - name: haproxy
+-      image: sles12/haproxy:__TAG__
++      image: sles12/haproxy:1.6.0
+       resources:
+         requests:
+           memory: 128Mi
+diff --git a/manifests/private.yaml b/manifests/private.yaml
+index 19e76ca..1ae9973 100644
+--- a/manifests/private.yaml
++++ b/manifests/private.yaml
+@@ -21,7 +21,7 @@ spec:
+   hostNetwork: False
+   initContainers:
+   - name: mariadb-secrets
+-    image: sles12/velum:__TAG__
++    image: sles12/velum:0.0
+     command: ["sh", "-c", "umask 377;
+                            if [ ! -f /infra-secrets/mariadb-root-password ]; then
+                              head -n 10 /dev/random | base64 | head -n 10 | tail -n 1 > /infra-secrets/mariadb-root-password;
+@@ -32,7 +32,7 @@ spec:
+       name: infra-secrets
+   containers:
+   - name: velum-mariadb
+-    image: sles12/mariadb:__TAG__
++    image: sles12/mariadb:10.0
+     env:
+     - name: MYSQL_DISABLE_REMOTE_ROOT
+       value: "true"
+diff --git a/manifests/public.yaml b/manifests/public.yaml
+index 0c4cab8..37d3051 100644
+--- a/manifests/public.yaml
++++ b/manifests/public.yaml
+@@ -21,7 +21,7 @@ spec:
+   hostNetwork: True
+   initContainers:
+   - name: mariadb-user-secrets
+-    image: sles12/mariadb:__TAG__
++    image: sles12/mariadb:10.0
+     command: ["/setup-mysql.sh"]
+     env:
+     - name: ENV
+@@ -35,7 +35,7 @@ spec:
+       name: setup-mysql
+       readOnly: True
+   - name: openldap-secrets
+-    image: sles12/velum:__TAG__
++    image: sles12/velum:0.0
+     command: ["sh", "-c", "umask 377;
+                            if [ ! -f /infra-secrets/openldap-password ]; then
+                              head -n 10 /dev/random | base64 | head -n 10 | tail -n 1 > /infra-secrets/openldap-password;
+@@ -45,7 +45,7 @@ spec:
+     - mountPath: /infra-secrets
+       name: infra-secrets
+   - name: saltapi-secrets
+-    image: sles12/velum:__TAG__
++    image: sles12/velum:0.0
+     command: ["sh", "-c", "umask 377;
+                            if [ ! -f /infra-secrets/saltapi-password ]; then
+                              head -n 10 /dev/random | base64 | head -n 10 | tail -n 1 > /infra-secrets/saltapi-password;
+@@ -55,7 +55,7 @@ spec:
+     - mountPath: /infra-secrets
+       name: infra-secrets
+   - name: salt-minion-key-generation
+-    image: sles12/salt-master:__TAG__
++    image: sles12/salt-master:2018.3.0
```

Comment:
Shouldn't we use `2018.3` only to be consistent? A very minor issue, I am just wondering if we should or not be strict on that.

Reply:
I think is a good point but this images are already released :) . We could fix that for the next images, but it is too late for the sles12 ones.

Reply:
I think the salt-images were always the odd-ones - the `2016` ones are already using this kind of format: `salt-master:2016.11.4`.

It also reflects the versioning our salt-team uses for our internal salt-releases (and I think a change from `2018.3.0` to let's say `2018.3.2` seems to be a *bigger* change already), so maybe keeping it might be worthwhile.